src/index.ts

```diff
@@ -76,9 +74,19 @@ export function optimize(
   peaks: Peak1D[];
   iterations: number;
 } {
-  if (!options.shape) {
-    options = { ...options, ...{ shape: { kind: 'gaussian' } } };
+  let peakList = JSON.parse(JSON.stringify(peakListInitial));
+  for (const peak of peakList) {
+    if (!peak.shape.kind) {
+      peak.shape.kind = options.defaultPeakShape || 'gaussian';
+    }
+    let kind = peak.shape.kind;
+    peak.shape = getShape1D({ kind: peak.shape.kind });
+    peak.shape.kind = kind;
+    peak.parameters = ['height', 'x', ...peak.shape.getParameters()];
+    peak.fromIndex = 0;
```

Comment:
This index is designed to be the position of the indexes in the array containing ALL the parameters to optimize.